microsoft-edge/devtools-guide-chromium/crash-analyzer/index.md

```diff
@@ -0,0 +1,338 @@
+---
+title: Crash analyzer tool
+description: How to use the Crash analyzer tool in DevTools.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: devtools
+ms.date: 05/09/2024
+---
+# Crash analyzer tool
+
+Use the **Crash analyzer** tool to analyze Microsoft Edge crashes.  The **Crash Analyzer** tool accelerates diagnosing JavaScript errors in production.  In the **Crash analyzer** tool, you can input a JavaScript stack trace, such as for non-fatal JavaScript exceptions, and then have your sourcemaps applied to the stack trace so that you can debug faster.  The **Crash Analyzer** tool enables you to take a minified JavaScript stack trace and rapidly work backwards to determine what lines of code caused the error.
+
+![The Crash analyzer tool](./index-images/crash-analyzer.png)
+<!-- todo: 
+maybe use page/file:
+https://devblogs.microsoft.com/oldnewthing/
+devtools://devtools/bundled/panels/elements/elements.js
+devtools://devtools/bundled/ui/components/tree_outline/tree_outline.js
+
+show stacktrace content in DevTools
+-->
+
+When a web app crashes or hangs unexpectedly, this can result in data loss and a poor user experience.  The **Crash analyzer** tool helps you identify the cause of these crashes.  Use the **Crash analyzer** tool to analyze the memory state of your web app at the time of a crash.  You can access The **Crash analyzer** tool from the **Memory** tool, which displays a list of recent crashes that occurred in your web app.  You can also trigger a crash dump manually by clicking the **Take snapshot** button.
```

Comment:
Regarding "You can also trigger a crash dump manually by clicking the **Take snapshot** button", is this true? The **Take snapshot** button in the **Memory** records a heap snapshot, not a crash stack trace.

Reply:
I'm also not seeing the list of recent crashes in the **Memory** tool. Do you know if this is intended to work?

My current understanding is that the expected workflow here is web devs collect stack traces from customers in production, and then use the Crash Analyzer tool to "beautify" them in DevTools.
So DevTools only does the analysis, not the collection.

If we have ways to also collect traces in DevTools, then that's great, but I would probably talk about them in a later section, as a "By the way, you can also collect local stack traces by ..." thing, if that makes sense. I would keep the introduction and first few section focused on the expected workflow.

Reply:
The statement "You can also trigger a crash dump manually by clicking the **Take snapshot** button" is copied from the blog post.  Maybe it's a misstatement in the blog post, and should be fixed here.
@captainbrosset

Reply:
added "my understanding" summary, and added hidden todo comments/questions:
```
When a web app crashes or hangs unexpectedly, 
this can result in data loss and a poor user experience.  
The **Crash analyzer** tool helps you identify the cause of these crashes.  
Use the **Crash analyzer** tool to analyze the memory state of your web app 
at the time of a crash.  
You can access The **Crash analyzer** tool from the **Memory** tool, 
which displays a list of recent crashes<!-- todo: true? --> that occurred in your web app.

As a web developer, you collect stack traces from customers in production, 
and then use the **Crash analyzer** tool to "beautify" the stack traces in DevTools.  
DevTools only does the analysis of the stack traces, not the collecting of stack traces.

As an alternative approach, you can collect local stack traces by triggering a crash dump manually 
by clicking the **Take snapshot** button in the **Memory** tool.<!-- todo: true? -->
```